app/com/linkedin/drelephant/exceptions/ExceptionFinder.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.linkedin.drelephant.exceptions;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.security.authentication.client.AuthenticatedURL;
+import org.apache.hadoop.security.authentication.client.AuthenticationException;
+import org.apache.log4j.Logger;
+import org.codehaus.jackson.JsonNode;
+import org.codehaus.jackson.map.ObjectMapper;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+
+public class ExceptionFinder {
+
+  private final Logger logger = Logger.getLogger(ExceptionFinder.class);
+  //private String url;
+  private Map<String, List<HadoopException>> exceptions;
+  private AzkabanClient _client;
+
+  public ExceptionFinder(String url) {
+    //this.url = url;
+    this.exceptions = new HashMap<String, List<HadoopException>>();
+    _client = new AzkabanClient(url);
+    _client.azkabanLogin("", ""); //Login using Azkaban Credentials
+    fetchAndAnalyzeLogs();
+  }
+
+  public void fetchAndAnalyzeLogs() {
+    final String jhistoryAddr = new Configuration().get("mapreduce.jobhistory.webapp.address");
+
+    LogAnalyzer analyzedFlowLog = analyzeAzkabanFlowLog();
+    Set<String> failedAzkabanJobs = analyzedFlowLog.getFailedSubEvents();
+    for (String azkabanJob : failedAzkabanJobs) {
+      LogAnalyzer analyzedAzkabanJobLog = analyzeAzkabanJobLog(azkabanJob);
+      List<HadoopException> exceptions = new ArrayList<HadoopException>();      // azkabanJob exceptions
+
+      Set<String> hadoopJobs = analyzedAzkabanJobLog.getFailedSubEvents();
+      for (String hadoopJob : hadoopJobs) {
+        String jhsURL = "http://" + jhistoryAddr + "/ws/v1/history/mapreduce/jobs/" + hadoopJob;
+        LogAnalyzer AnalyzedMRLog = analyzeHadoopJobOverview(jhsURL);
+        if (AnalyzedMRLog != null) {
+          HadoopException mrException = new HadoopException();
+          if (AnalyzedMRLog.getException().getExceptionChain() != null) {
+            //mr job failed for reason other than task failure
+            ExceptionLoggingEvent mrJobException = AnalyzedMRLog.getException();
```

Comment:
You need to use the LoggingEvent class or some wrapper of it instead of creating your own exception class. We need that anyway to generate the hash.